var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -36,18 +36,19 @@ class PyNumpy(PythonPackage):
     homepage = "http://www.numpy.org/"
     url      = "https://pypi.io/packages/source/n/numpy/numpy-1.9.1.tar.gz"
 
+    version('1.12.0', '33e5a84579f31829bbbba084fe0a4300',
+            url="https://pypi.io/packages/source/n/numpy/numpy-1.12.0.zip")
```

Comment:
I'll wait for one more `.zip` exclusive release before I make the call and switch it all to `.zip`. Good point about making people re-download/re-sync mirrors, I didn't think of that.

> I believe the extrapolator (at least my one) will base the URL on the most recent URL in the list.

See #2737, which is actually explained in #2725. Basically, if I add a newer version to this package and don't specify a URL, it will extrapolate from the `url` for `1.12.0`, not the package-level `url`. This might trip up new users. It sure tripped me up!